### PR TITLE
fix(mobile): Fixes an issue where deleted images would still appear in the gallery

### LINKED ISSUE #2748: [BUG] Removed photo is still cached and visible when swipe back to it in mobile app
### The bug

In mobile app, when opening photo (like "full screen" mode) and removing it (from album or everywhere), I swiping back and see it again.

### The OS that Immich Server is running on

Windows 10

### Version of Immich Server

v1.60.0

### Version of Immich Mobile App

v1.60.0 build.83

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:release
    command: ["start-server.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    depends_on:
      - redis
      - database
      - typesense
    restart: always

  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:release
    command: ["start-microservices.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    depends_on:
      - redis
      - database
      - typesense
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    image: ghcr.io/immich-app/immich-machine-learning:release
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  immich-web:
    container_name: immich_web
    image: ghcr.io/immich-app/immich-web:release
    env_file:
      - .env
    restart: always

  typesense:
    container_name: immich_typesense
    image: typesense/typesense:0.24.0
    environment:
      - TYPESENSE_API_KEY=${TYPESENSE_API_KEY}
      - TYPESENSE_DATA_DIR=/data
    logging:
      driver: none
    volumes:
      - tsdata:/data
    restart: always

  redis:
    container_name: immich_redis
    image: redis:6.2
    restart: always

  database:
    container_name: immich_postgres
    image: postgres:14
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      PG_DATA: /var/lib/postgresql/data
    volumes:
      - pgdata:/var/lib/postgresql/data
    restart: always

  immich-proxy:
    container_name: immich_proxy
    image: ghcr.io/immich-app/immich-proxy:release
    environment:
      # Make sure these values get passed through from the env file
      - IMMICH_SERVER_URL
      - IMMICH_WEB_URL
    ports:
      - 2283:8080
    depends_on:
      - immich-server
    restart: always

volumes:
  pgdata:
  model-cache:
  tsdata:
```

### Your .env content

```Shell
###################################################################################
# Database
###################################################################################

DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_PASSWORD=postgres
DB_DATABASE_NAME=immich

# Optional Database settings:
# DB_PORT=5432




###################################################################################
# Redis
###################################################################################

REDIS_HOSTNAME=immich_redis

# Optional Redis settings:
# REDIS_PORT=6379
# REDIS_DBINDEX=0
# REDIS_PASSWORD=
# REDIS_SOCKET=





###################################################################################
# Upload File Config
###################################################################################

UPLOAD_LOCATION=./photos

###################################################################################
# Typesense
###################################################################################
TYPESENSE_API_KEY=some-random-text
# TYPESENSE_ENABLED=false
# TYPESENSE_URL uses base64 encoding for the nodes json.
# Example JSON that was used:
# [
#      { 'host': 'typesense-1.example.net', 'port': '443', 'protocol': 'https' },
#      { 'host': 'typesense-2.example.net', 'port': '443', 'protocol': 'https' },
#      { 'host': 'typesense-3.example.net', 'port': '443', 'protocol': 'https' },
#  ]
# TYPESENSE_URL=ha://WwogICAgeyAnaG9zdCc6ICd0eXBlc2Vuc2UtMS5leGFtcGxlLm5ldCcsICdwb3J0JzogJzQ0MycsICdwcm90b2NvbCc6ICdodHRwcycgfSwKICAgIHsgJ2hvc3QnOiAndHlwZXNlbnNlLTIuZXhhbXBsZS5uZXQnLCAncG9ydCc6ICc0NDMnLCAncHJvdG9jb2wnOiAnaHR0cHMnIH0sCiAgICB7ICdob3N0JzogJ3R5cGVzZW5zZS0zLmV4YW1wbGUubmV0JywgJ3BvcnQnOiAnNDQzJywgJ3Byb3RvY29sJzogJ2h0dHBzJyB9LApd

###################################################################################
# Reverse Geocoding
#
# Reverse geocoding is done locally which has a small impact on memory usage
# This memory usage can be altered by changing the REVERSE_GEOCODING_PRECISION variable
# This ranges from 0-3 with 3 being the most precise
# 3 - Cities > 500 population: ~200MB RAM
# 2 - Cities > 1000 population: ~150MB RAM
# 1 - Cities > 5000 population: ~80MB RAM
# 0 - Cities > 15000 population: ~40MB RAM
####################################################################################

# DISABLE_REVERSE_GEOCODING=false
# REVERSE_GEOCODING_PRECISION=3

#DISABLE_REVERSE_GEOCODING=true


####################################################################################
# Alternative Service Addresses - Optional
#
# This is an advanced feature for users who may be running their immich services on different hosts.
# It will not change which address or port that services bind to within their containers, but it will change where other services look for their peers.
# Note: immich-microservices is bound to 3002, but no references are made
####################################################################################

IMMICH_WEB_URL=http://immich-web:3000
IMMICH_SERVER_URL=http://immich-server:3001
IMMICH_MACHINE_LEARNING_URL=http://immich-machine-learning:3003


####################################################################################
# WEB - Optional
#
# Custom message on the login page, should be written in HTML form.
# For example:
# PUBLIC_LOGIN_PAGE_MESSAGE="This is a demo instance of Immich.<br><br>Email: <i>demo@demo.de</i><br>Password: <i>demo</i>"
####################################################################################

PUBLIC_LOGIN_PAGE_MESSAGE=

####################################################################################
# Alternative API's External Address - Optional
#
# This is an advanced feature used to control the public server endpoint returned to clients during Well-known discovery.
# You should only use this if you want mobile apps to access the immich API over a custom URL. Do not include trailing slash.
# NOTE: At this time, the web app will not be affected by this setting and will continue to use the relative path: /api
# Examples: http://localhost:3001, http://immich-api.example.com, etc
####################################################################################

#IMMICH_API_URL_EXTERNAL=http://localhost:3001
```

### Reproduction steps

```bash
1. In mobile app, open some uploaded photo (I didn't test with photos that are not uploaded)
2. Press "trash" button to remove it
3. Swipe to the right to back on the previous photo. Removed photo visible here.
4. (optionally) Press "trash" button again. Error message should occur.
```

### Additional information

_No response_

### LINKED ISSUE #4272: [BUG] mobile: deleted asset reappears while viewing
### The bug

Deleted asset reappears again after its removal while viewing assets.

### The OS that Immich Server is running on

Fedora 38

### Version of Immich Server

v1.79.1

### Version of Immich Mobile App

v1.79.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
not relevant
```

### Your .env content

```Shell
not relevant
```

### Reproduction steps

```bash
1. view asset, make sure there is at least one more asset on the left (previous)
2. delete asset, next asset is shown in the background while system is asking for confirmation of asset removal
3. swipe to right to go to previous asset
4. removed asset is shown again, however with thumbnail resolution (because the original is really deleted)
5. swipe to right again, now the correct asset is shown
6. when viewing is exited, removed asset disappears from the grid
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by martyfuhry**
- Sends the renderList directly to the gallery view
- The gallery view can interact with the renderList

This is a workaround for now. Soon I will refactor the gallery and timelines to watch the same render list and we can remove some of this. In the meanwhile, I was able to simplify quite a bit of code and this is a slight improvement over what we were doing.

### CODE REVIEW

File: mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart [hunk 1]
```diff
@@ -311,4 +329,11 @@ class RenderList {
     GroupAssetsBy groupBy,
   ) =>
       _buildRenderList(assets, null, groupBy);
+
+  /// Deletes an asset from the render list and clears the buffer
+  /// This is only a workaround for deleted images still appearing in the gallery
+  Future<void> deleteAsset(Asset deleteAsset) async{
+    allAssets?.remove(deleteAsset);
+    _buf.clear();
+  }
```
Comment (martyfuhry) [inline mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart:338]: @fyfrey Does this look OK to you?

File: mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart [hunk 2]
```diff
@@ -56,6 +56,24 @@ class RenderList {
...
+        .map(
+          (a) => RenderAssetGridElement(
+            RenderAssetGridElementType.assets,
+            date: a.fileCreatedAt,
+          ),
+        )
+        .toList();
+    return RenderList(elements, null, assets);
+  }
+
```
Comment (martyfuhry) [inline mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart:76]: This is only used in one place (map_page) which I didn't test...

Comment (fyfrey) [reply mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart:76]: We already got the static method fromAssets. Is that not enough?

File: mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart (hunk 1 above)
Comment (fyfrey) [reply mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart:338]: To be sure: also set bufOffset=0.

Does the method need to be async at all?

Comment (martyfuhry) [reply mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart]: Nope, it doesn't. I also set bufOffset.

**Comment by martyfuhry** [comment]
Fixed the merge conflicts, ready for review once more.

**Review by fyfrey — APPROVED** [review]
LGTM

**Comment by declan8010** [comment]
@alextran1502 So you think we can get this in the next release, it's been around a while

**Comment by waclaw66** [comment]
Could this be merged already?

**Pull request closed without merging**